src/stdlib/Json.ts

```diff
@@ -0,0 +1,127 @@
+import { AssociativeArray } from "../brsTypes/components/AssociativeArray";
+import { BrsArray } from "../brsTypes/components/BrsArray";
+import { Interpreter } from "../interpreter";
+import { Literal } from "../parser/Expression";
+import { randomBytes } from "crypto";
+
+import {
+    BrsBoolean,
+    BrsInvalid,
+    BrsString,
+    BrsType,
+    Callable,
+    Float,
+    Int32,
+    Int64,
+    ValueKind
+} from "../brsTypes";
+
+function brsValueOf(x: any): any {
+    if (x === null) { return BrsInvalid.Instance; }
+    let t: string = typeof x;
+    switch (t) {
+    case "boolean":
+        return BrsBoolean.from(x);
+    case "string":
+        return new BrsString(x);
+    case "number":
+        if (Number.isInteger(x)) {
+            return x >= -2_147_483_648 && x <= 2_147_483_647 ? new Int32(x) : new Int64(x);
+        }
+        return new Float(x);
+    case "object":
+        if (Array.isArray(x)) {
+            return new BrsArray(x.map(brsValueOf));
+        }
+        return new AssociativeArray(
+            Object.getOwnPropertyNames(x).map((k: string) => {
+                return {
+                    name: brsValueOf(k),
```

Comment:
An associative array's keys must be strings in brightscript, but there's no easy way to guarantee that `brsValueOf(someString)` will return a `BrsString` at compile-time.  Switching `brsValueOf` to return `BrsType` complains here because `BrsBoolean`, `BrsInvalid`, etc. aren't valid brightscript strings (naturally).  Since we know as humans that these need to be strings, and we know at compile time that `k` is a string, it's safe to just construct a `new BrsString(k)` here!  Plus, it's really straightforward 😄

Reply:
done